### PR TITLE
Send SQL case updates to elasticsearch

### PR DESCRIPTION
**Pull request by czue**
sneaking this in under the :gun: 

@snopoke cc @benrudolph @proteusvacuum 

:fish: :office:

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/processors/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from .interface import PillowProcessor
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/processors/__init__.py]: F401 'PillowProcessor' imported but unused

File: corehq/ex-submodules/pillowtop/processors/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+from .sample import NoopProcessor, LoggingProcessor
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/processors/__init__.py]: F401 'LoggingProcessor' imported but unused

Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/processors/__init__.py]: F401 'NoopProcessor' imported but unused

File: corehq/ex-submodules/pillowtop/processors/interface.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from abc import ABCMeta, abstractmethod
+
+
+class PillowProcessor(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def process_change(self, pillow_instance, change, do_set_checkpoint):
+        pass
+
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/processors/interface.py]: W391 blank line at end of file

File: corehq/ex-submodules/pillowtop/processors/__init__.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+from .elastic import ElasticProcessor
```
Comment (dimagimon) [inline corehq/ex-submodules/pillowtop/processors/__init__.py]: F401 'ElasticProcessor' imported but unused

File: corehq/pillows/case.py [hunk 5]
```diff
@@ -52,8 +61,42 @@ def calc_meta(cls):
...
+        if doc_ret.get("user_id"):
+            doc_ret["owner_id"] = doc_ret["user_id"]
+    return doc_ret
+
+
+def prepare_sql_case_json_for_elasticsearch(sql_case_json):
+    prepped_case = transform_case_for_elasticsearch(sql_case_json)
+    # todo: these are required for consistency with couch representation, figure out how best to deal with it
+    prepped_case['doc_type'] = 'CommCareCase'
+    prepped_case['_id'] = prepped_case['case_id']
```
Comment (czue) [inline corehq/pillows/case.py]: wasn't sure what our long term strategy should be for this property coming out of elastic

**Comment by snopoke** [comment]
Looks great. 

Notes for me: things still to do:

- [ ] hard deletions
- [ ] soft deletions for case

File: corehq/pillows/case.py (hunk 5 above)
Comment (snopoke) [inline corehq/pillows/case.py]: Seems fine to leave as is for now. If we want to change it later it will be the same amount of work.

**Comment by czue** [comment]
:white_check_mark:

**Comment by benrudolph** [comment]
:sake::bomb:

**Pull request merged**